plenum/test/node_catchup/test_catchup_from_unequal_nodes_without_reasking.py

```diff
@@ -32,19 +32,24 @@ def test_catchup_from_unequal_nodes_without_reasking(looper,
                                                      txnPoolNodeSet,
                                                      sdk_pool_handle,
                                                      sdk_wallet_client):
-    # Node numbers are selected so that test fails when using old algo
-    lagged_node_1 = txnPoolNodeSet[2]
-    lagged_node_2 = txnPoolNodeSet[-1]
+    lagged_node_1 = txnPoolNodeSet[-1]
+    lagged_node_2 = txnPoolNodeSet[-2]
+    normal_nodes = [node for node in txnPoolNodeSet
+                    if node not in [lagged_node_1, lagged_node_2]]
+    normal_stashers = [node.nodeIbStasher for node in normal_nodes]
+
     with delay_rules_without_processing(lagged_node_1.nodeIbStasher, delay_3pc()):
         sdk_send_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle, sdk_wallet_client, 2)
 
         with delay_rules_without_processing(lagged_node_2.nodeIbStasher, delay_3pc()):
             sdk_send_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle, sdk_wallet_client, 7)
-            ensure_all_nodes_have_same_data(looper, set(txnPoolNodeSet) - {lagged_node_2, lagged_node_1},
-                                            custom_timeout=30)
+            ensure_all_nodes_have_same_data(looper, normal_nodes, custom_timeout=30)
+
+            # Make sure that we will receive cons proof from lagging node before all other nodes
+            start_delaying(normal_stashers, lsDelay(delay=0.5))
```

Comment:
Should we have more explicit check, that we received ConsProofs from all the nodes?

Reply:
I don't think we need such check because:
* we don't actually want to wait for cons proofs from ALL nodes
* all we care about is that we receive cons proof from lagging node before normal nodes, so it is not discarded if normal nodes provide quorum of equal cons proofs faster